### PR TITLE
Switch cookie library, update to match new API

### LINKED ISSUE #31: Remove vendor/cookie.js
Cookie.js (`assets/vendor/js/cookie.js`) is using an incompatible GPLv3 license. Rewrite it from scratch under MIT ensuring that no code breaks in the front-end. The API may change (it may have to), but any JavaScript that calls it should reflect that change.

This is a good first issue for new contributors.

### PR DESCRIPTION
**Pull request by waynegerard**
Switch the cookie library to one with a more permissive license.

Also change call sites to reflect the changed API (get vs. getItem, set
vs. setItem, expire vs. removeItem).

Fixes #31

### CODE REVIEW

**Comment by waynegerard** [comment]
One issue this brings up is that everything under `public/*` is under `.gitignore`, so the `main.js` file isn't quite up-to-date unless you run the asset compilation step yourself.

File: assets/js/app.js [hunk 1]
```diff
@@ -25,34 +25,34 @@ abot.signout = function(ev) {
...
+	    Cookies.get("authToken") != null) {
 		return true
 	}
-	// If the user isn't logged in, ensure we clean out all cookies.
-	cookie.setItem("id", null)
-	cookie.setItem("email", null)
-	cookie.setItem("issuedAt", null)
-	cookie.setItem("scopes", null)
-	cookie.setItem("authToken", null)
+	// If the user isn't logged in, ensure we clean out all Cookiess.
```
Comment (egtann) [inline assets/js/app.js]: Small typo :)

File: assets/js/app.js [hunk 2]
```diff
@@ -25,34 +25,34 @@ abot.signout = function(ev) {
...
+	Cookies.set("id", null)
```
Comment (egtann) [inline assets/js/app.js]: Should we be setting them to `null` here or expiring them like above?

**Comment by egtann** [comment]
> One issue this brings up is that everything under public/* is under .gitignore, so the main.js file isn't quite up-to-date unless you run the asset compilation step yourself.

That's a great point. In development, should we recompile the assets on each page load?

File: assets/js/app.js (hunk 1 above)
Comment (waynegerard) [inline assets/js/app.js]: Heh whoops! Fixed.

File: assets/js/app.js (hunk 2 above)
Comment (waynegerard) [inline assets/js/app.js]: That's fair. I would also learn towards expiring them, since that seems the safest.

**Comment by waynegerard** [comment]
Recompiling assets on each page load seems ideal in dev, since you'd want to see JS changes immediately.

I feel like it's maybe outside the purview of this PR, but I can certainly draft something together for that.

**Comment by egtann** [comment]
re: this PR. Looks great! Could we rebase this into 1 commit? (You can `--force` push to your fork/branch and that'll modify this PR directly.)   Your first commit message is perfect: `Switch cookie library, update to match new API`. Then we'll be ready to merge!

re: compile assets. Looking at the code, right now we do run a CompileAssets function in the Index handler (outside of production), so we should be set. For reference, CompileAssets is defined in `core/boot.go`, and it's called in both `core/boot.go` (on server boot, even in production) and `core/handlers.go` (on page reload outside of production)

**Comment by waynegerard** [comment]
Rebased! 

And that's good to hear about CompileAssets. Is that a recent change? It didn't seem to be quite working when I looked at this a week or so back.

Could be just a weird one-off thing.

**Comment by egtann** [comment]
CompileAssets isn't a recent change, so it's entirely possible it isn't working! :)

**Pull request merged**